src/v/config/property.h

```diff
@@ -593,4 +595,27 @@ class enum_property : public property<T> {
     std::vector<T> _values;
 };
 
+class retention_property final : public property<std::chrono::milliseconds> {
+public:
+    using property<std::chrono::milliseconds>::property;
+
+    void set_value(std::any v) override {
+        auto value = std::any_cast<std::chrono::milliseconds>(std::move(v));
```

Comment:
can time be negative? wouldn't it make sense to cast it as an int?

Reply:
`std::duration`s can be negative, they're backed by a `long long`.